walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,115 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        
+        // no meeting time available 
+        if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+            return Arrays.asList();
+        }
+        
+        // no other clashing events
+        if (events.isEmpty()) {
+            return Arrays.asList(TimeRange.WHOLE_DAY);
+        }
+
+        // flag to check mutual exclusivity of attendee list
+        boolean distinctAttendees = true;
```

Comment:
I got confused by the name of this boolean and comment to mean if the attendees are distinct or mutually exclusive with each other and there are duplicates in the list. You may wanna rename and update comment. Longer name is ok when needed.

Reply:
Whoops, I see your point. I'll rename it to something more representative of its meaning.